### PR TITLE
cmake build with gcc4.8

### PR DESCRIPTION
**Pull request by lalitb**
We support experimental C++11 on gcc4.8 , but cmake build is missing for that. Adding a GitHub action for that.

### CODE REVIEW

**Review by maxgolov — APPROVED** [review]

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -21,6 +21,23 @@ jobs:
     - name: run prometheus exporter tests
       run: ./ci/do_ci.sh cmake.exporter.prometheus.test
 
+  cmake_gcc_48_test:
+    name: Legacy CMake
```
Comment (g-easy) [inline .github/workflows/ci.yml:25]: How about "CMake gcc 4.8" so it's clear that isn't not a legacy version of CMake that's being tested?

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -21,6 +21,23 @@ jobs:
...
+        sudo ./ci/install_gcc48.sh
+    - name: setup cmake
+      run: |
+        sudo CC=/usr/bin/gcc-4.8 CXX=/usr/bin/g++-4.8 ./ci/setup_cmake.sh
+    - name: run tests
+      run: ./ci/do_ci.sh cmake.legacy.test
+      env:
+        CC: /usr/bin/gcc-4.8
+        CXX: /usr/bin/g++-4.8
   cmake_test_cxx20:
```
Comment (g-easy) [inline .github/workflows/ci.yml:41]: Add a blank line for consistency please.

File: ci/do_ci.sh [hunk 3]
```diff
@@ -30,6 +30,16 @@ elif [[ "$1" == "cmake.c++20.test" ]]; then
   make
   make test
   exit 0
+elif [[ "$1" == "cmake.legacy.test" ]]; then
+  cd "${BUILD_DIR}"
+  rm -rf *
+  cmake -DCMAKE_BUILD_TYPE=Debug  \
+        -DCMAKE_CXX_FLAGS="-Werror" \
+        -DCMAKE_CXX_STANDARD=11 \
```
Comment (g-easy) [inline ci/do_ci.sh:38]: It doesn't work without the explicit C++11 option?

File: ci/do_ci.sh [hunk 4]
```diff
@@ -30,6 +30,16 @@ elif [[ "$1" == "cmake.c++20.test" ]]; then
...
+        "${SRC_DIR}"
+  make
+  make test
+  exit 0;
```
Comment (g-easy) [inline ci/do_ci.sh:42]: Don't need the semicolon.

File: .github/workflows/ci.yml (hunk 1 above)
Comment (lalitb) [reply .github/workflows/ci.yml]: Done

File: ci/do_ci.sh (hunk 3 above)
Comment (lalitb) [reply ci/do_ci.sh:38]: No, gcc4.8 default compilation mode is `-std=c++0x`. We need to make it `-std=c++11` explicitly.

**Review by g-easy — APPROVED** [review]

**Pull request merged**